public-api/src/main/java/org/pmiops/workbench/service/AchillesAnalysisService.java

```diff
@@ -0,0 +1,742 @@
+                } else {
+                    if (stratum == 1) {
+                        missingResult = achillesMapper.makeCopyAchillesResult(aa.getAnalysisId(), missingGender, null, null, null, null, null,null, 20L, 20L);
+                    } else if (stratum == 2) {
+                        missingResult = achillesMapper.makeCopyAchillesResult(aa.getAnalysisId(), conceptId, missingGender, null, null, null, null,null, 20L, 20L);
+                    } else if (stratum == 3) {
+                        missingResult = achillesMapper.makeCopyAchillesResult(aa.getAnalysisId(), conceptId, null, missingGender, null, null, null,null, 20L, 20L);
+                    }
+                }
+                missingResult.setAnalysisStratumName(genderStratumNameMap.get(missingGender));
+                aa.getResults().add(missingResult);
+            }
+        }
+        aa.setResults(aa.getResults().stream().filter(ar -> ar.getAnalysisStratumName() != null).collect(Collectors.toList()));
+    }
+
+    public void addAgeStratum(Analysis aa, String conceptId, List<DbAchillesResult> ehrCountResults, int stratum){
+        Set<String> uniqueAgeDeciles = new TreeSet<String>();
+        for(AchillesResult ar: aa.getResults()){
+            String analysisStratumName=ar.getAnalysisStratumName();
+            if (analysisStratumName == null || analysisStratumName.equals("")) {
+                if (stratum == 2 && ar.getStratum2() != null && !ar.getStratum2().equals("0")) {
+                        uniqueAgeDeciles.add(ar.getStratum2());
+                        ar.setAnalysisStratumName(ageStratumNameMap.get(ar.getStratum2()));
+                } else if (stratum == 4 && ar.getStratum4() != null && !ar.getStratum4().equals("0")) {
+                        uniqueAgeDeciles.add(ar.getStratum4());
+                        ar.setAnalysisStratumName(ageStratumNameMap.get(ar.getStratum4()));
+                }
+            }
+        }
+        aa.setResults(aa.getResults().stream().filter(ar -> ar.getAnalysisStratumName() != null).collect(Collectors.toList()));
+        if(uniqueAgeDeciles.size() < 8){
+            if (aa.getAnalysisId() == CommonStorageEnums.analysisIdFromName(AnalysisIdConstant.EHR_AGE_COUNT_ANALYSIS_ID)) {
+                Set<String> completeAgeDeciles = new TreeSet<String>(Arrays.asList(new String[] {"2", "3", "4", "5", "6", "7", "8", "9"}));
+                completeAgeDeciles.removeAll(uniqueAgeDeciles);
+                for(String missingAgeDecile: completeAgeDeciles){
+                    List<DbAchillesResult> ehrAgeCountResults = null;
+                    try {
+                        ehrAgeCountResults  = ehrCountResults.stream().filter(ar -> ar.getStratum4().equals(missingAgeDecile)).collect(Collectors.toList());
+                    } catch(NullPointerException npe) {
```

Comment:
This catch is empty. I would add a null check in your filter above a remove this.

Reply:
Changed this. Thanks.